greengrass_defender_agent/agent.py

```diff
@@ -106,13 +108,26 @@ def publish_metrics(ipc_client, config_changed, metrics_collector, sample_interv
 def main():
     # Get the ipc client
     ipc_client = ipc_utils.IPCUtils()
-    try:
-        ipc_client.connect()
-    except Exception as e:
-        config.logger.error(
-            "Exception occurred during the creation of an IPC client: {}".format(e)
-        )
-        exit(1)
+
+    need_retry = True
+    retry_time = config.INITIAL_RETRY_INTERVAL_SECONDS
+
+    while need_retry:
```

Comment:
Seems it's retrying infinitely? I wonder if we should set a max retry count since we can't tell what causes the exception and in some error scenarios it does need to exit.

Reply:
Make sense. Let's discuss the max retry count offline.

Reply:
I'm thinking that we should also retry publishing for network issues or not break when it fails publishing? Maybe not to re-raise the exception in publish_metrics() https://github.com/aws-greengrass/aws-greengrass-device-defender/blob/a804600917a08ce3ada5eab0744bb1f8c97857c8/greengrass_defender_agent/agent.py#L103 ?